app/models/manageiq/providers/amazon/inventory/persister/shared/cloud_collections.rb

```diff
@@ -0,0 +1,84 @@
+module ManageIQ::Providers::Amazon::Inventory::Persister::Shared::CloudCollections
+  extend ActiveSupport::Concern
+
+  included do
```

Comment:
Do we need this? It's all just method definitions, we shouldn't need this? Unless I'm missing something :-)

Reply:
We have to replace Amazon's InventoryCollectionDefault classes with something else.
1) initialize_inventory_collection() method is more readable - this mixin is the reason why you wrote you like this approach / it simply defines "add_orchestration_stacks") method etc.
2) Amazon's ```Persister::CloudManager``` and ```Persister::TargetCollection``` (partially with NetworkManager) - and not only amazon's persisters - shares provider specific collections with slightly different settings. 
---
So without this mixin, you have to 1) repeat yourself in code and 2) have code uglier

Reply:
Or if you're talking about ```ActiveSupport::Concern``` - why not? :)

Reply:
I mean, do we need the ```included do``` ? :-)

Reply:
Yeah, I think we can remove it from most (all) of the places. It will shift it to left and make it more readable. :-)